Runtime/Core/ApiGatewayManagement/AmazonGameLiftClientWrapper.cs

```diff
@@ -42,17 +57,191 @@ public async Task<SearchGameSessionsResponse> SearchGameSessions(SearchGameSessi
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                throw;
+            }
         }
 
-        public async Task<DescribeGameSessionsResponse> DescribeGameSessions(DescribeGameSessionsRequest request)
+        private async Task RegisterCompute()
         {
-            return await _amazonGameLiftClient.DescribeGameSessionsAsync(request);
+            try
+            {
+                var ipAddress = Dns.GetHostEntry(Dns.GetHostName())
+                    .AddressList
+                    .FirstOrDefault(ip => ip.AddressFamily == AddressFamily.InterNetwork)?
+                    .ToString() ?? "0.0.0.0";
+                
+                var registerComputeRequest = new RegisterComputeRequest()
+                {
+                    ComputeName = _gameLiftClientSettings.ComputeName,
+                    FleetId = _gameLiftClientSettings.FleetID,
+                    IpAddress = ipAddress,
+                    Location = _gameLiftClientSettings.FleetLocation
+                };
+                var registerComputeResponse = await RegisterCompute(registerComputeRequest);
+                    
+                _gameLiftClientSettings.WebSocketUrl = registerComputeResponse.Compute.GameLiftServiceSdkEndpoint;
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
         }
+
+        private async Task CreateCustomLocation()
+        {
+            try
+            {
+                ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;
+                var listLocationsResponse = await ListLocations(new ListLocationsRequest
+                {
+                    Filters = new List<string>{ "CUSTOM" }
+                });
+                
+                var foundLocation = listLocationsResponse.Locations.FirstOrDefault(l => l.LocationName.ToString() == _gameLiftClientSettings.FleetLocation);
+
+                if (foundLocation == null)
+                {
+                    var createLocationResponse = await CreateLocation(new CreateLocationRequest()
+                    {
+                        LocationName = _gameLiftClientSettings.FleetLocation
+                    });
+                    
+                    if (createLocationResponse.HttpStatusCode == HttpStatusCode.OK)
+                    {
+                        Console.WriteLine($"Created Custom Location {_gameLiftClientSettings.FleetLocation}");
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.Message);
+            }
+        }
+        
+        private async Task CreateFleetCommand()
+        {
+            try
+            {
+                var createFleetRequest = new CreateFleetRequest
+                {
+                    ComputeType = ComputeType.ANYWHERE,
```

Comment:
Same as above comment: FleetType, FleetLocation should be parameters passed into this method.